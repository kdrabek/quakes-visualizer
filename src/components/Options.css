.options {
  grid-area: options;
}

.container {
  background: #f3f6f8;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  padding: 30px;
}

.date {
  order: 0;
  width: 30%;
  align-self: center;
}

.mag {
  order: 1;
  width: 60%;
}

.react-datepicker__input-container > input {
  width: 100%;
}
