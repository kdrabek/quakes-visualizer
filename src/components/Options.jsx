import React from 'react';
import DatePicker from 'react-datepicker';
import moment from 'moment';
import Slider from 'react-rangeslider';
import { PropTypes } from 'prop-types';
import classNames from 'classnames/bind';
import { range, fromPairs } from 'lodash';
import { withState, withHandlers, compose } from 'recompose';

import 'react-datepicker/dist/react-datepicker.css';
import styles from './Options.css';

const cx = classNames.bind(styles);
const optionsDate = cx({ date: true });
const optionsMagnitude = cx({ mag: true });
const options = cx({ options: true });
const optionsContainer = cx({ container: true });

const propTypes = PropTypes && {
  text: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
const defaultProps = {};

const magnitudeLabels = fromPairs(range(1, 11).map(v => [v, v.toString()]));

export const Options = props => (
  <div className={options}>
    <div className={optionsContainer}>
      <span className={optionsDate}>
        Pick a date:
        <DatePicker
          disableFuture
          locale="en-gb"
          className="date"
          dateFormat="YYYY-MM-DD"
          label="Choose a date"
          minDate={moment('1900-01-01')}
          maxDate={moment()}
          selected={moment(props.date)}
          onChange={props.onDateChange}
          showYearDropdown
          dropdownMode="select"
        />
      </span>
      <span className={optionsMagnitude}>
        Select min magnitude:
        <Slider
          min={1}
          max={10}
          labels={magnitudeLabels}
          value={props.magnitude}
          tooltip={false}
          onChange={props.onMagnitudeChange}
        />
      </span>
    </div>
  </div>
);

Options.displayName = 'Options';
Options.propTypes = propTypes;
Options.defaultProps = defaultProps;

export default compose(
  withState('date', 'setDate', moment()),
  withState('magnitude', 'setMagnitude', 5),
  withHandlers({
    onDateChange: props => (date) => {
      props.setDate(date);
      props.handleOnClick(date, props.magnitude);
    },
    onMagnitudeChange: props => (magnitude) => {
      props.setMagnitude(magnitude);
      props.handleOnClick(props.date, magnitude);
    },
  }),
)(Options);
