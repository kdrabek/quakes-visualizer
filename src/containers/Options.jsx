import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import handler from '../handlers/earthquake';
import Options from '../components/Options';

const propTypes = PropTypes && {
  actions: PropTypes.object.isRequired,
  earthquakes: ImmutablePropTypes.list,
};

export class OptionsContainer extends PureComponent {
  render() {
    return <Options handleOnClick={this.props.actions.getEarthquakes} />;
  }
}

OptionsContainer.propTypes = propTypes;
OptionsContainer.displayName = 'OptionsContainer';

export function mapStateToProps(state, props) {
  return { ...props };
}

export function mapDispatchToProps(dispatch) {
  const boundActions = bindActionCreators({ ...handler.actions }, dispatch);
  return { actions: boundActions };
}

export default connect(mapStateToProps, mapDispatchToProps)(OptionsContainer);
