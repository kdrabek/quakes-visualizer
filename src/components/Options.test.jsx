import { shallow } from 'enzyme';
import Options from './Options';

jest.mock('moment', () => () => '2018–04–09T12:34:56+00:00');

describe('Options component', () => {
  const props = {};

  it('should match snapshot', () => {
    const wrapper = shallow(<Options {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
