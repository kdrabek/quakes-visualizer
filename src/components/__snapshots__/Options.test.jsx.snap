// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options component should match snapshot 1`] = `
<form
  autoComplete="off"
>
  <WithStyles(FormControl)>
    <WithStyles(InputLabel)
      htmlFor="magnitude"
    >
      Magnitude
    </WithStyles(InputLabel)>
    <WithStyles(Select)
      inputProps={
        Object {
          "id": "magnitudeSelect",
          "name": "magnitude",
        }
      }
      onChange={[Function]}
      value={5}
    >
      <WithStyles(MenuItem)
        key="1"
        value={1}
      >
        1
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="2"
        value={2}
      >
        2
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="3"
        value={3}
      >
        3
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="4"
        value={4}
      >
        4
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="5"
        value={5}
      >
        5
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="6"
        value={6}
      >
        6
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="7"
        value={7}
      >
        7
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="8"
        value={8}
      >
        8
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="9"
        value={9}
      >
        9
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="10"
        value={10}
      >
        10
      </WithStyles(MenuItem)>
    </WithStyles(Select)>
  </WithStyles(FormControl)>
  <DatePickerWrapper
    autoOk={false}
    disableFuture={true}
    format="YYYY-MM-DD"
    label="Choose a date"
    minDate="1900-01-01"
    onChange={[Function]}
    returnMoment={true}
    utils={
      Object {
        "formatNumber": [Function],
        "getCalendarHeaderText": [Function],
        "getDatePickerHeaderText": [Function],
        "getDateTimePickerHeaderText": [Function],
        "getDayText": [Function],
        "getHourText": [Function],
        "getMeridiemText": [Function],
        "getMinuteText": [Function],
        "getMonthNumber": [Function],
        "getNextMonth": [Function],
        "getPreviousMonth": [Function],
        "getStartOfMonth": [Function],
        "getWeekArray": [Function],
        "getWeekdays": [Function],
        "getYear": [Function],
        "getYearText": [Function],
        "setYear": [Function],
      }
    }
    value={2018-03-25T16:16:51.674Z}
  />
  <WithStyles(Button)
    color="primary"
    onClick={[Function]}
  >
    Submit
  </WithStyles(Button)>
</form>
`;
